app/src/main/java/com/example/android/example/TrailerViewHolder.java

```diff
@@ -0,0 +1,59 @@
+package com.example.android.example;
+
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.example.android.example.Data.Trailer;
+import com.squareup.picasso.Picasso;
+
+import java.util.List;
+
+import static com.example.android.example.Adapters.TrailerAdapter.YT_BASE_PATH;
+
+public class TrailerViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+    private String youtube_thumbail = "https://img.youtube.com/vi/";
+
+    ImageView trailerImageView;
+    TextView videoTitleTextView;
+    TextView videoTypeTextView;
+    public ImageView shareIcon;
+    List<Trailer> trailer;
+    Context context;
+
+    public TrailerViewHolder(View itemView, final Context context,final List<Trailer> trailer) {
+        super(itemView);
+        trailerImageView = (ImageView) itemView.findViewById(R.id.iv_detail_trailer);
+        videoTitleTextView = itemView.findViewById(R.id.tv_video_title);
+        videoTypeTextView = itemView.findViewById(R.id.tv_video_type);
+        shareIcon = itemView.findViewById(R.id.share_icon);
+        this.trailer=trailer;
+
+        this.context = context;
+        itemView.setOnClickListener(this);
+    }
+
+
+    public void bind(int listIndex) {
+        String trailerThumbnailUrl = youtube_thumbail + trailer.get(listIndex).getTrailer_thumbnail() + "/maxresdefault.jpg";
```

Comment:
Once you remove the list reference from this class. You can change the method definition to `bind(Trailer trailer)`.
So your adapter will pass the trailer object rather than the index.

Reply:
You can set the `onClickListener` in `onBind()`